R/utils_url.R

```diff
@@ -176,7 +176,16 @@ url_process_error <- function(url, req, headers) {
 url_port <- function(host, port, process, verbose = TRUE) {
   spin <- cli::make_spinner()
   if_any(verbose, spin$spin(), NULL)
-  while (!pingr::is_up(destination = host, port = port)) {
+  .is_up <- function(host, port) {
+    con <- try(suppressWarnings(socketConnection(host, port, open = "r")), silent = TRUE)
```

Comment:
IMO a slightly clearer style here is something like:

```R
tryCatch(
  {
    con <- suppressWarnings(socketConnection(host, port, open = "r"))
    close(con)
    TRUE
  },
  error = function(cnd) {
    FALSE
  }
)```

(And note that this still won't work in webR)

Reply:
> (And note that this still won't work in webR)

That seems okay because `tar_watch()` itself is only meant to be run locally. `shinylive` apps will be using the module in `tar_watch_ui()` and `tar_watch_server()`, which do not use `pingr`. (Although until today `tar_watch_ui()` and `tar_watch_server()` did still enforce all the same dependencies as `tar_watch()`, including `pingr`, which was an oversimplification.)